network/p2p/readSubscription.go

```diff
@@ -78,7 +79,9 @@ func (r *readSubscription) receiveLoop(wg *sync.WaitGroup) {
 
 		var msg message.Message
 		// convert the incoming raw message payload to Message type
+		p := binstat.EnterTimeVal("~3net:wire>1protobuf2message", "", int64(len(rawMsg.Data)))
 		err = msg.Unmarshal(rawMsg.Data)
+		binstat.Leave(p)
```

Comment:
do we want to keep binstat always on for this method?

Reply:
@ramtinms I guess that depends what you mean by 'always on'? :-)

By default `binstat` is 'disabled' because the env var `BINSTAT_ENABLE` is not set. Disabled means that the enter and leave functions are still called but then there's something like `if disabled { return }`. So minimal overhead but it's not like C/C++ where the functions can be 'compiled away' via the pre-processor so it's like they were never there. Do you know a better way to minimize the overhead?

What is the overhead if `binstat` is 'enabled'? This currently depends on the `BINSTAT_LEN_WHAT` env var. In the above example then the first argument is `"~3net:wire>1protobuf2message"` which means that unless `BINSTAT_LEN_WHAT` says differently then the bin will be shortened to only `~3net`... which creates less bins to be updated and dumped. However, `BINSTAT_LEN_WHAT="~net=99"` means use up to 99 characters of the bin and if there are multiple bins starting with `~3net` then this will result in more bins and more `binstat` output. Lastly, if `BINSTAT_LEN_WHAT="~net=0"`is used then this means do not use any bins for `~3net` stats. The `binstat` enter function uses a little bit of CPU to figure out that it's doing nothing. And the leave function becomes just a return.

So as you can see, there are multiple ways to control the `binstat` hose of bins and stats :-)

Reply:
AFAIK if we want `binstat` to be 'compiled away' then it would be necessary to create a specific and unique node app binary with it 'compiled in'. I wasn't sure how to do that on a function by function / use case by use case basis, so I didn't so far. My understanding is that it would be in theory possible to write a Golang compiler add-on to do this... and I wouldn't mind investigating this in the future as such a technique could be generally useful. But so far I have not found a good tutorial on how to do it... so it's not a 'low hanging fruit' :-(